www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,454 @@
+
+## Chef Upgrade Lab Requirements
+
+### System Architecture Requirements
+
+The Chef Lab makes some basic assumptions about your existing system:
+
+* Two Chef Servers
+  -  A Chef Server running some older version
+  -  A newly installed Chef Infra Server running the latest software, set up and configured for knife, but otherwise empty
+* One or more nodes
+  - Running Chef Infra Client 12-15
+  - Minimum 512 MB RAM
+  - [Recommended](https://docs.chef.io/chef_system_requirements/#chef-infra-client) 5GB space
+  - Bootstrapped to the older Chef Server
+  - `sudo` permissions on the node
+  - SSH connectivity between the nodes and the administrator/developer workstation
+* An administrator/developer workstation
+  - 64-bit architecture
+  - Minimum 4 GB RAM
+  - Recommended 10 GB of available disk space for installing Chef Workstation and running the Chef Upgrade Lab
+
+### Software Requirements
+
+* Meet the [platform and system requirements](https://docs.chef.io/workstation/install_workstation/) for Chef Workstation
+* Install or upgrade to the Chef Workstation [latest verision](https://downloads.chef.io/chef-workstation)
+* Any [supported Ruby version](https://www.ruby-lang.org/en/downloads/)
+
+Chef does not prescribe any specific editor. However, the [Chef Infra extension](https://marketplace.visualstudio.com/items?itemName=chef-software.Chef) for [Visual Studio Code](https://code.visualstudio.com/) features several code generators and helpful features, such as running Cookstyle each time you save a recipe.
+
+### Server Backup
+
+We recommend performing a backup before starting any server upgrade process.
+Follow the [Chef Infra Server Backup](https://docs.chef.io/runbook/server_backup_restore/) documentation before starting your Upgrade Lab.
+
+### Chef Infrastructure Requirements
+
+#### Credentials
+
+Your credentials are set up using [knife profiles](https://docs.chef.io/workstation/knife_setup/#knife-profiles). This allows you to keep your keys in a `credentials` file, and makes switching between credentials easier.
+
+For example, in `.chef/credentials`:
+
+```toml
+[old-server]
+client_name = "myuser"
+chef_server_url = "https://old-chef-server.dev/organizations/my-org"
+client_key = """
+-----BEGIN RSA PRIVATE KEY-----
+MMM+some+key+goes+here+MMM
+-----END RSA PRIVATE KEY-----
+"""
+[new-server]
+client_name = "myuser"
+chef_server_url = "https://new-chef-server.dev/organizations/my-org"
+client_key = """
+-----BEGIN RSA PRIVATE KEY-----
+MMM+another+key+goes+here+MMM
+-----END RSA PRIVATE KEY-----
+"""
+```
+
+#### Connectivity
+
+* You have a user key for both of the Chef Infra Servers
+* You can connect to both Chef Infra Servers from your development workstation.
+
+Verify connectivity by running a knife command against each server and receive a reasonable response:
+
+```shell
+chef exec knife user list --profile old-server my-user
+chef exec knife user list --profile new-server my-user
```

Comment:
Both commands here should have the `my-user` portion on the next line - it is example output